src/middleware/validator/generic.validator.ts

```diff
@@ -12,7 +12,7 @@ export function validate(req: Request, res: Response, next: NextFunction) {
 }
 
 export function idAsInt(req: Request, res: Response, next: NextFunction) {
-  if (!req.params.id) return next()
+  if (!req.params.id) return next() // TODO: Shouldn't this return a 400?
```

Comment:
We should just fix this now. There's very very few usages of this, and it'll be easier to refactor now than later. I went through them all and I think that it should just do something like this instead of what it's doing now

```
if (!req.params.id) throw new Error("Missing Id on Id required request") 
```

Or something like that. 

My guess is the reason I wrote it this way was if we wanted to add this as global middleware. If this were global middleware it'd check that `id` if in the route would be required to be a number everywhere. If that were the case then we would want this to just allow this to pass through to the request if no Id was present. The upside to this would be we would only have to add this middleware in one place; the downside would be that it'd be requiring any route we use the exact string `id` in would require that I'd to be a number (which is probably fine but idk). 

So yea if want this global, make it global and leave this calling next if no Id is present. If we want to add this just where we use it then we can change this to throw if no Id is there.

Reply:
That makes sense. I like it the way it's setup since it's explicit that some paths require an id instead just moving on when there's no id.

Is there a reason why you suggest throwing an error here, and the check for id being an int immediately returns a 400? Unless there's a reason for the difference, I think both should throw an error and let the error handler process the response. And if I'm understanding express error handling, should both be something like

`next(new Error("message"))`

Finally, will we ever have a path that has 2 ids (ie. courseId and assignmentId)? Would we want to make this check more general to prepare for that? Like take the name of the id as a param and return the middleware?

Reply:
Ya know I just left another comment saying we should return 400 vs throw an error 😂

Because the global middleware handles errors afaik to the client it should be the same regardless. However, actively returning a response is technically a faster code path because it breaks the chain and returns early. 

As for middleware with multiple ids :thinking: I don't know if we can write that into this middleware simply because they'll have different names. 

Example 
```
/users/:id/submissions/:submissionId
```

In the above example we're using 2 ids, `id` and `submissionId` (keep in mind in a real scenario like this I'd probably want to use `userId` to be more explicit), because the id name is dynamic we'd have to keep arbitrarily adding those ids here which idk if that makes sense.

What we could do is make a more dynamic function that takes a property name? Then we could pass in this middleware and the property of the id that we think should be there and be a number?

Either way, I think that's a future problem.

Reply:
Future problem, but easy enough of a solution to add so we don't have to refactor later. I committed a solution and tested it with Postman. Take a look and if we want to use this I'll update all the paths.